pkg/tsdb/cloudwatch/time_series_query_test.go

```diff
@@ -136,3 +137,268 @@ func TestTimeSeriesQuery(t *testing.T) {
+						"namespace": "",
+						"metricName": "",
+						"region": "us-east-2",
+						"id": "query id",
+						"statistic": "Maximum",
+						"period": "1200",
+						"hide": false,
+						"matchExact": true,
+						"refId": "A"
+					}`),
+			},
+		},
+	})
+
+	assert.NoError(t, err)
+	assert.Equal(t, "query id", resp.Responses["A"].Frames[0].Name)
+}
+
+func Test_QueryData_executeTimeSeriesQuery_no_alias_provided_frame_name_depends_on_dimension_values_and_matchExact(t *testing.T) {
+	origNewCWClient := NewCWClient
+	t.Cleanup(func() {
+		NewCWClient = origNewCWClient
+	})
+	var cwClient fakeCWClient
+	NewCWClient = func(sess *session.Session) cloudwatchiface.CloudWatchAPI {
+		return &cwClient
+	}
+	cwClient = fakeCWClient{
+		GetMetricDataOutput: cloudwatch.GetMetricDataOutput{
+			MetricDataResults: []*cloudwatch.MetricDataResult{
+				{StatusCode: aws.String("Complete"), Id: aws.String("query id"), Label: aws.String("response label"),
+					Values: []*float64{aws.Float64(1.0)}, Timestamps: []*time.Time{{}}},
+			},
+		},
+	}
+	im := datasource.NewInstanceManager(func(s backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		return datasourceInfo{}, nil
+	})
+	executor := newExecutor(im, newTestConfig(), &fakeSessionCache{})
+
+	t.Run("frame name is label when isInferredSearchExpression and not isMultiValuedDimensionExpression", func(t *testing.T) {
+		testCasesReturningLabel := map[string]struct {
+			dimensions string
+			matchExact bool
+		}{
+			"with specific dimension, matchExact false": {dimensions: `"dimensions": {"InstanceId": ["some-instance"]},`, matchExact: false},
+			"with wildcard dimension, matchExact false": {dimensions: `"dimensions": {"InstanceId": ["*"]},`, matchExact: false},
+			"with wildcard dimension, matchExact true":  {dimensions: `"dimensions": {"InstanceId": ["*"]},`, matchExact: true},
+			"without dimension, matchExact false":       {dimensions: "", matchExact: false},
+		}
+		for name, tc := range testCasesReturningLabel {
+			t.Run(name, func(t *testing.T) {
+				resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+					PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+					Queries: []backend.DataQuery{
+						{
+							RefID:     "A",
+							TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+							JSON: json.RawMessage(fmt.Sprintf(`{
+						"type":      "timeSeriesQuery",
+						"metricQueryType": 0,
+						"metricEditorMode": 0,
+						"namespace": "",
+						"metricName": "",
+						%s
+						"region": "us-east-2",
+						"id": "query id",
+						"statistic": "Maximum",
+						"period": "1200",
+						"hide": false,
+						"matchExact": %t,
+						"refId": "A"
+					}`, tc.dimensions, tc.matchExact)),
+						},
+					},
+				})
+
+				assert.NoError(t, err)
+				assert.Equal(t, "response label", resp.Responses["A"].Frames[0].Name)
```

Comment:
nit: you can save "response label" in a variable so this is easier to read. Something like: `assert.Equal(t,  metricDataResultLabel, resp.Responses["A"].Frames[0].Name)`

Reply:
Done, forgot your variable suggestion while I was writing it but did something similar. Let me know if it works for you